File: designPattern/php/AbstractFactory.php
<?php

interface Human
{
    public function say();
}

class Man implements Human
{
    public function say()
    {
        echo '男人<br/>';
    }
}

class Woman implements Human
{
    public function say()
    {
        echo '女人<br/>';
    }
}

interface Factory
{
    public function createMan();
}

class ManFactory implements Factory
{
    public function createMan()
    {
        return new Man();
    }
}

class WomanFactory implements Factory
{
    public function createMan()
    {
        return new Woman();
    }
}

$man = new ManFactory();
$woman = new WomanFactory();
$manSay = $man->createMan();
$womanSay = $woman->createMan();
$manSay->say();
$womanSay->say();

File: designPattern/AbstractFactory.py
# -*- coding: utf-8 -*-

class Human:
    def say(self):
        pass

class Man(Human):
    def say(self):
        print('男人')

class Woman(Human):
    def say(self):
        print('女人')

class AbstractFactory:
    def creatMan(self):
        pass

class ManFactory(AbstractFactory):
    def creatMan(self):
        return Man()

class WomanFactory(AbstractFactory):
    def creatMan(self):
        return Woman()

if __name__ == "__main__":
    testFactory = ''
    sex = input('请输入性别：')
    if sex == '男人':
        testFactory = ManFactory()
    elif sex == '女人':
        testFactory = WomanFactory()
    else:
        print('操作错误')
        exit(0)

    sextest = testFactory.creatMan()
    sextest.say()
